### PR TITLE
[SYCL] Release notes for June'20 DPCPP implementation update

### PR DESCRIPTION
**Pull request by romanovvlad**
Increase major version of libsycl.so library as well

### CODE REVIEW

File: sycl/ReleaseNotes.md [hunk 1]
```diff
@@ -1,3 +1,157 @@
...
+    as a `sycl::detail::bit_cast` [e3da4ef]
+  - Introduced the Level Zero plugin which enables SYCL working on top of Level0
+    API. Interoperability is not supportet yet [d32da99]
+  - Implemented [parallel_for simplification extension](doc/extensions/ParallelForSimpification) [13fe9fb]
+  - Implemented [SYCL_INTEL_enqueue_barrier extension](doc/extensions/EnqueueBarrier/enqueue_barrier.asciidoc) [da6bfd0]
+  - Implemented [SYCL_INTEL_accessor_simplification extension](https://github.com/intel/llvm/pull/1498) [1f76efc]
+  - Implemented OpenCL interoperability API following [SYCL Generalization proposal](https://github.com/KhronosGroup/SYCL-Shared/blob/master/proposals/sycl_generalization.md) [bae0639]
+
+## Improvements
+### SYCL Frontend and driver changes
```
Comment (bader) [inline sycl/ReleaseNotes.md:20]: I think we should rename this section to "SYCL compiler" and next section to "SYCL library".

File: sycl/ReleaseNotes.md [hunk 2]
```diff
@@ -1,3 +1,157 @@
...
+  - A SYCL program will be aborted now if program linking is requested when
+    using L0 plugin. This is done because L0 doesn't support program linking
+    [d4a5b71]
+  - Added a diagnostic on attempt to use `sycl::program::set_spec_constant` when
+    the program is already in compiled or linked state [e2e3d3d]
+  - Improved `sycl::stream` class implementation on the device side in order to
+    reduce local memory consumption [b838f0e]
+
+### Documentation
+  - Added [a table](doc/extensions/README.md) with DPC++ extensions statuses
```
Comment (bader) [inline sycl/ReleaseNotes.md:73]:
```suggestion
  - Added [a table](doc/extensions/README.md) with DPC++ extensions status
```

File: sycl/ReleaseNotes.md [hunk 3]
```diff
@@ -1,3 +1,157 @@
...
+    been improved [9aa5029]
+  - The [SYCL_INTEL_sub_group extension document](doc/extensions/SubGroup/SYCL_INTEL_sub_group.asciidoc)
+    has been updated [010f112]
+  - Render user API classes on a dedicated page [98b6ee4]
+
+## Bug fixes
+### SYCL Frontend and driver changes
+  - Fixed device code compile options passing which could lead to
+    `CL_INVALID_COMPILER_OPTIONS` error [57bad9e]
+  - Fixed a problem which caused problems with creating a queue for FPGA device
```
Comment (bader) [inline sycl/ReleaseNotes.md:86]:
```suggestion
  - Fixed a problem with creating a queue for FPGA device
```

File: sycl/ReleaseNotes.md (hunk 2 above)
Comment (s-kanaev) [reply sycl/ReleaseNotes.md:73]: Done.

File: sycl/ReleaseNotes.md (hunk 3 above)
Comment (s-kanaev) [reply sycl/ReleaseNotes.md:86]: Done.

File: sycl/ReleaseNotes.md (hunk 1 above)
Comment (s-kanaev) [reply sycl/ReleaseNotes.md]: Done

**Pull request merged**